### PR TITLE
feat: Implement orb discovery

### LINKED ISSUE #181: Implement orb discovery
https://trustbloc.github.io/did-method-orb/#discovery

### PR DESCRIPTION
**Pull request by fqutishat**
closes #181

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by bstasyszyn — APPROVED** [review]

File: pkg/restapi/discovery/controller.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package discovery
```
Comment (sandrask) [inline pkg/restapi/discovery/controller.go:6]: discovery should probably be in separate package, not under restapi

File: pkg/restapi/internal/common/support/httphandler.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+package support
+
+import (
+	"net/http"
+
+	"github.com/trustbloc/sidetree-core-go/pkg/restapi/common"
+)
+
+// Handler http handler for each controller API endpoint.
+type Handler interface {
```
Comment (sandrask) [inline pkg/restapi/internal/common/support/httphandler.go:15]: probably no need for this common handler

File: pkg/restapi/discovery/controller.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/discovery/controller.go]: done

File: pkg/restapi/internal/common/support/httphandler.go (hunk 2 above)
Comment (fqutishat) [reply pkg/restapi/internal/common/support/httphandler.go]: done

**Review by sandrask — APPROVED** [review]

**Pull request merged**